### PR TITLE
Formula parser & alias cleanup & rich-based help

### LINKED ISSUE #17: change yaml.dump() to yaml.safe_dump() in Cargo.build_argument_list
Since the other end expects standard YaML, better to use safe_dump.

### LINKED ISSUE #20: implement expression parsing for substitutions
As discussed in https://github.com/caracal-pipeline/stimela2/discussions/26#discussioncomment-2442396

### PR DESCRIPTION
**Pull request by o-smirnov**
Sorry @SpheMakh, this is a big gulp. I know I keep beating the drum for small feature branches, but since you're ignoring my PRs, they just snowball into major ones.

### CODE REVIEW

File: scabha/types.py [hunk 1]
```diff
@@ -1,9 +1,33 @@
+import os.path
```
Comment (sphemakh) [inline scabha/types.py:1]: elmo

File: tests/test_parsing.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        a. b
+        """)
+
+#    a = expr.parse_string("((a.x))")
+    
+
+
+
+if __name__ == "__main__":
+    test_parser()
```
Comment (sphemakh) [inline tests/test_parsing.py]: add new line at end

**Review by SpheMakh — CHANGES_REQUESTED** [review]
very minor changes

**Comment by o-smirnov** [comment]
OK @SpheMakh please approve and merge.

**Pull request merged**